src/core/core/rpc_engine.cpp

```diff
@@ -144,7 +144,12 @@ namespace dsn {
                 req->server_address.group_address()->leader_forward();
             }
 
-            call->set_delay(delay_ms);
+            // the "delay_ms" is simulator's net_delay_milliseconds,
+            // the "call->delay_milliseconds()" is user specified delay time to dispatch response task,
```

Comment:
I don't get the need for dispatch delay: if it is for delaying the rpc call, then we should use lpc to delay the call; if it is for delaying the response task execution, then the point should be about network delay or fault injection, which is done by our tools using join-points.  If we really need this in certain scenario, I would suggest adding a new method called ```add_delay``` if needed to make the semantic clear.